controllers/authorController.js

```diff
@@ -15,35 +19,196 @@ exports.author_list = function(req, res) {
 
 // Display detail page for a specific Author.
 exports.author_detail = function(req, res) {
-    res.send('NOT IMPLEMENTED: Author detail: ' + req.params.id);
+    
+  async.parallel({
+    author: function(callback) {
+        Author.findById(req.params.id)
+          .exec(callback)
+    },
+    authors_blogposts: function(callback) {
+      BlogPost.find({ 'author': req.params.id },'title summary')
+      .exec(callback)
+    },
+}, function(err, results) {
+    if (err) { return next(err); } // Error in API usage.
+    if (results.author==null) { // No results.
+        let err = new Error('Author not found');
+        err.status = 404;
+        return next(err);
+    }
+    // Successful, so render.
+    res.render('author_detail', { title: 'Author Detail', author: results.author, author_blogposts: results.authors_blogposts } );
+});
 };
 
 // Display Author create form on GET.
-exports.author_create_get = function(req, res) {
-    res.send('NOT IMPLEMENTED: Author create GET');
+exports.author_create_get = function(req, res, next) {       
+  res.render('author_form', { title: 'Create Author'});
 };
 
+
 // Handle Author create on POST.
-exports.author_create_post = function(req, res) {
-    res.send('NOT IMPLEMENTED: Author create POST');
-};
+exports.author_create_post = [
+
+  // Validate and sanitise fields.
+  body('first_name').trim().isLength({ min: 1 }).escape().withMessage('First name must be specified.')
+      .isAlphanumeric().withMessage('First name has non-alphanumeric characters.'),
+  body('family_name').trim().isLength({ min: 1 }).escape().withMessage('Family name must be specified.')
+      .isAlphanumeric().withMessage('Family name has non-alphanumeric characters.'),
+  body('date_of_birth', 'Invalid date of birth').optional({ checkFalsy: true }).isISO8601().toDate(),
+  body('date_of_death', 'Invalid date of death').optional({ checkFalsy: true }).isISO8601().toDate(),
+  
+  // Process request after validation and sanitization.
+  (req, res, next) => {
+
+      //provide edited dates
+      
+      let author_temp = {
+        first_name: req.body.first_name,
+        family_name: req.body.family_name,
+        date_birth_form: DateTime.fromJSDate(req.body.date_of_birth).toISODate(),
```

Comment:
I've removed the toDate() conversion in the validation/sanitation section which was not needed.

I'm creating date_birth_form (which converts the ISO8601 date to ISODate) so that when updating the author (or if there is an error at validation and the form gets re-rendered) the date displays correctly in the form.  If i leave it as a ISO8601 date (which mongo stores in the model and I don't seem to be able to change from reading the documentation) then there is an error when passing that to the form and the date is blank.  

It does seem like I still have a lot of dates/virtual dates though (even once I get rid of the date of death)...  Am I missing something?